File: ruby/acronym/acronym.rb
module Acronym
  VERSION = 1

  def self.abbreviate(text)
    text
      .scan(/\b(\w)|[a-z]([A-Z])/)
      .map do |first_letter, camel_case|
        (first_letter || camel_case).upcase
      end.join
  end
end

File: elixir/acronym/acronym.exs
defmodule Acronym do

  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(string) :: String.t()
  def abbreviate(string) do
    ~r/\b([A-Za-z])|[a-z]([A-Z])/
    |> Regex.scan(string)
    |> Enum.map(fn (match) -> List.last(match) end)
    |> Enum.join
    |> String.upcase
  end
end
